### PR TITLE
[interop] C++ destination record should be destroyed before being cop…

### PR DESCRIPTION
**Pull request by hyp**
…ied into during assignWithCopy / assignWithTake

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

File: test/Interop/Cxx/class/Inputs/copy-move-assignment.h
Comment (zoecarver) [inline test/Interop/Cxx/class/Inputs/copy-move-assignment.h:1]: Nit, non-blocking: the other special member tests are in `Cxx/value-witness-table`.

**Review by zoecarver — APPROVED** [review]

**Comment by zoecarver** [comment]
Thanks Alex. This LGTM.

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test macos platform

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Pull request merged**